tests/cases/collection_test.py

```diff
@@ -172,3 +180,33 @@ def testCollectionAccess(self):
         access = resp.json
         self.assertEqual(access['users'][0]['id'], str(self.user['_id']))
         self.assertEqual(access['users'][0]['level'], AccessType.WRITE)
+        coll = self.model('collection').load(self.collection['_id'], force=True)
+        folder1 = self.model('folder').load(folder1['_id'], force=True)
+        folder2 = self.model('folder').load(folder2['_id'], force=True)
+        self.assertEqual(coll['public'], True)
+        self.assertEqual(folder1['public'], False)
+
+        # Update the collection recursively to public
+        resp = self.request(
+            path='/collection/%s/access' % coll['_id'], method='PUT', params={
+                'access': json.dumps(obj),
+                'public': True,
+                'recurse': True,
+                'progress': True
+            }, user=self.admin)
+        self.assertStatusOk(resp)
+        coll = self.model('collection').load(coll['_id'], force=True)
+        folder1 = self.model('folder').load(folder1['_id'], force=True)
+        folder2 = self.model('folder').load(folder2['_id'], force=True)
+        self.assertEqual(coll['public'], True)
+        self.assertEqual(folder1['public'], True)
+        self.assertEqual(folder2['public'], True)
+        self.assertEqual(folder1['access'], coll['access'])
+        self.assertEqual(folder1['access'], folder2['access'])
+        self.assertEqual(folder2['access'], {
+            'users': [{
+                'id': self.user['_id'],
+                'level': AccessType.WRITE
+            }],
+            'groups': []
+        })
```

Comment:
On something as important as setting access control I'd like a bit more testing.

I'd like to see a test case where there is a tree that is public (extending this existing case would work fine) and gets set to private, and then testing that multiple levels in the tree inherit the private access.  I'd also like to see in that same case a deletion of access, this could be as simple as taking the existing test access object and removing the user write access, setting this access on the tree recursively, then checking that this has been removed at multiple levels.